src/main/java/msjava2lesson/Main.java

```diff
@@ -1,7 +1,397 @@
 package msjava2lesson;
 
+import msjava2lesson.moveZeroes.Solution;
+import msjava2lesson.reverselinkedlist.ListNode;
+import msjava2lesson.validparantheses.ValidParenthesesSolution;
+
+import java.util.Arrays;
+
 public class Main {
+
     public static void main(String[] args) {
-        System.out.println("Hello World!");
+        taskReverseLinkedList();
+        taskValidParentheses();
+        taskMoveZeros();
+    }
+
+    private static void linkedListPrint(ListNode node) {
+        if (node != null) {
+            System.out.print("Head:");
+        }
+
+        while (node != null) {
+            System.out.print(node.val + " -> ");
+            node = node.next;
+        }
+        System.out.println("No pointer, the end...");
+    }
+
+    private static void taskValidParentheses() {
+
+        ValidParenthesesSolution validParenthesesSolution = new ValidParenthesesSolution();
+
+        String test = "()";
+        System.out.println("String " + test + " is" + (validParenthesesSolution.isValid(test)?"":" not") + " valid.");
+
+        test = "([])";
+        System.out.println("String " + test + " is" + (validParenthesesSolution.isValid(test)?"":" not") + " valid.");
+
+        test = "([{}])";
+        System.out.println("String " + test + " is" + (validParenthesesSolution.isValid(test)?"":" not") + " valid.");
+
+        test = "()[]{}";
+        System.out.println("String " + test + " is" + (validParenthesesSolution.isValid(test)?"":" not") + " valid.");
+
+        test = "(]";
+        System.out.println("String " + test + " is" + (validParenthesesSolution.isValid(test)?"":" not") + " valid.");
+
+        test = "[]]"; // "" // "asdasd"
+        System.out.println("String " + test + " is" + (validParenthesesSolution.isValid(test)?"":" not") + " valid.");
+
+        // For  "" and "asdasd" will throw an exception.
+        System.out.println();
+        System.out.println();
+    }
```

Comment:
Duplicated code code be moved to a separate method
```suggestion
        private static void taskValidParentheses() {
          ValidParenthesesSolution validParenthesesSolution = new ValidParenthesesSolution();
          List<String> testCases = List.of(
              "()", 
              "([])", 
              "([{}])", 
              "()[]{}", 
              "(]", 
              "[]]"
          );
  
          for (String test : testCases) {
              printValidationResult(test, validParenthesesSolution);
          }
    }

    private static void printValidationResult(String test, ValidParenthesesSolution solution) {
        System.out.println("String " + test + " is" + (solution.isValid(test) ? "" : " not") + " valid.");
    }

```